pkg/operation/seed/seed.go

```diff
@@ -1293,3 +1311,26 @@ func deleteIngressController(ctx context.Context, c client.Client) error {
 		&corev1.Service{ObjectMeta: metav1.ObjectMeta{Name: "nginx-ingress-k8s-backend", Namespace: v1beta1constants.GardenNamespace}},
 	)
 }
+
+func comparePriorityClassValue(ctx context.Context, k8sClient client.Client, priorityClassName string, valueToCompare int) (int, error) {
+	pc := &schedulingv1.PriorityClass{}
+	if err := k8sClient.Get(ctx, kutil.Key(priorityClassName), pc); client.IgnoreNotFound(err) != nil && !meta.IsNoMatchError(err) {
+		return 0, err
+	}
+	return valueToCompare - int(pc.Value), nil
+}
+
+func deletePriorityClassIfValueNotTheSame(ctx context.Context, k8sClient client.Client, priorityClassName string, valueToCompare int) error {
+	result, err := comparePriorityClassValue(ctx, k8sClient, priorityClassName, valueToCompare)
+	if err != nil {
+		return err
+	}
+	if result == 0 {
+		return nil
+	}
```

Comment:
Could you simplify this to just 
1. get the resource
1. If there is error
   1. If it does not exist, return without errors
   1. return the error
1. if the value is the same, return without errors
1. delete the resource

What do you think?

Reply:
ok